src/oasis-indexer/api.ts

```diff
@@ -1,38 +1,61 @@
 /** @file Wrappers around generated API */
 
-import axios, { AxiosResponse } from 'axios'
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios'
 import { paraTimesConfig } from '../config'
 import * as generated from './generated/api'
 import BigNumber from 'bignumber.js'
 import { UseQueryOptions } from '@tanstack/react-query'
 import { Layer } from './generated/api'
 import { getEthAccountAddress } from '../app/utils/helpers'
+import { Network } from '../types/network'
 
 export * from './generated/api'
 export type { RuntimeEvmBalance as Token } from './generated/api'
 
 declare module './generated/api' {
   export interface Transaction {
+    network: Network
     layer: Layer
   }
   export interface RuntimeTransaction {
+    network: Network
     layer: Layer
   }
   export interface Block {
+    network: Network
     layer: Layer
   }
   export interface RuntimeBlock {
+    network: Network
     layer: Layer
   }
   export interface Account {
+    network: Network
     layer: Layer
   }
   export interface RuntimeAccount {
+    network: Network
     layer: Layer
     address_eth?: string
   }
 }
 
+function wrapWithNetwork<P extends Array<any>, Q>(f: (...args: P) => Q): (net: Network, ...args: P) => Q {
+  return (net, ...args) => {
+    console.log('Should use', net)
+    return f(...args)
+  }
+}
+
+const baseUrls: Partial<Record<Network, string>> = {
+  testnet: process.env.REACT_TESTNET_APP_API,
+}
+
+const getBaseUrlParamFor = (network: Network): Partial<AxiosRequestConfig<any>> => {
+  const baseURL = baseUrls[network]
+  return baseURL ? { baseURL } : {}
```

Comment:
This doesn't mean an error, this means that this specific network doesn't require a specific baseURL override parameter, since the default one will suffice. (since we are on mainnet.)

Reply:
I have just added some explicit error checking for missing config.